src/olympia/versions/models.py

```diff
@@ -576,6 +576,8 @@ def watch_source(old_attr=None, new_attr=None, instance=None, sender=None,
 def update_status(sender, instance, **kw):
```

Comment:
Yes, it's called on `post_save` and `post_delete`. I had to clear the cached property and reload the add-on in there to make sure it wasn't overwriting something another signal modified (some tests broke without)